app/scripts/controllers/projects_controller.js

```diff
@@ -0,0 +1,31 @@
+(function(){
+  'use strict';
+
+  angular
+    .module('app.controllers')
+    .controller('ProjectsController', ProjectsController);
+
+  ProjectsController.$inject = ['$location', 'DataFactory', '$firebaseObject', '$routeParams'];
+
+  function ProjectsController($location, DataFactory, $firebaseObject, $routeParams){
+    var project = this;
```

Comment:
@yeuem1vannam 

Em nghĩ là ở trong controller thì gán `vm = this` hoặc `_this = this` thôi chứ mỗi cái một tên riêng này có vẻ khó theo dõi.